### DISCUSSION TITLE
Extracting Pattern RecExpr from a conditional applier?

### DISCUSSION THREAD

**Discussion by cowardsa** [egraphs-good/egg#179] [Q&A]
I'm having difficulty extracting the pattern representing the rhs of my rewrite rule if it is a conditional rewrite rule.
For standard rewrites, using applier.get_pattern_ast() works great! It doesn't seem to return anything for a conditional applier though?

**Comment by mwillsey** [comment]
We could implement it for conditional applier, but we could only use the underlying applier's AST and ignore the condition. I didn't think it was prudent to return the AST when it was the "whole" AST, so I left it unimplemented.

**Reply by cowardsa** [reply]
Yep that solution works (just tried it) - is it undesirable to go into the main branch?

**Reply by mwillsey** [reply]
I think we should leave this out of the main branch for now. Getting a "partial" ast back feels like it could be a foot-gun in many contexts.
